src/App.js

```diff
@@ -1,5 +1,60 @@
+import {
+  stringifyToJSON,
+  readLineAsync,
+  validatePositiveNumberArray,
+  convertCharacterClassRegex,
+  sumArray,
+  printResult,
+  convertNumberArray,
+} from './utils.js';
+
 class App {
-  async run() {}
+  static DEFAULT_SEPARATOR = ',:';
+  static CUSTOM_SEPARATOR_REGEXP = /\/\/(.+)\\n/;
+
+  async run() {
+    const userInput = await readLineAsync();
+    const processedInput = this.processInput(userInput);
+    const sum = this.calculateSum(processedInput);
+
+    printResult(sum);
+  }
+
+  processInput(input) {
+    if (input === '') return []; // 빈 문자열은 예외로 0을 리턴한다.
+
+    const stringified = stringifyToJSON(input);
+    const separator = this.getSeparator(stringified);
+    const content = this.extractContent(stringified);
+
+    return this.splitContent(content, separator);
+  }
```

Comment:
```suggestion
  beforeProcess(input) {
    if (input === '') return []; // 빈 문자열은 예외로 0을 리턴한다.

    const stringified = stringifyToJSON(input);
    
    return { separator: this.getSeparator(stringified), content: this.extractContent(stringified) };
  }
  processInput(input) {
    const { content, separator } = beforeProcess(input);

    return this.splitContent(content, separator);
  }
```

processInput도 충분히 좋은 함수지만 더 작은 함수를 만든다는 관점에서 조금 더 분리해보면 어떨까요?

Reply:
오, 이 방식은 해본 적이 없는데 코드가 더 무거워지면 말씀해주신 방식으로 전처리와 처리를 분리하는 것이 큰 효과를 발휘할 거 같아요.

좋은 아이디어입니다 👍